lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1503,6 +1503,21 @@ class DecomposeAten_ToCopyOp : public OpRewritePattern<Aten_ToCopyOp> {
 };
 } // namespace
 
+namespace {
+// Decompose `aten.new_empty` op into `aten.empty.memory_format` op.
+class DecomposeAtenNewEmptyOp : public OpRewritePattern<AtenNewEmptyOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenNewEmptyOp op,
+                                PatternRewriter &rewriter) const override {
+    Value noneVal = rewriter.create<ConstantNoneOp>(op.getLoc());
+    rewriter.replaceOpWithNewOp<AtenEmptyMemoryFormatOp>(
+        op, op.getType(), op.size(), op.dtype(), op.layout(), op.device(),
+        op.pin_memory(), /*memory_format=*/noneVal);
```

Comment:
Yes, you are correct about the definition. But the thing you are asking for is handled in the `RefineType.cpp` itself:
https://github.com/llvm/torch-mlir/blob/46ecb84e57714d9d936181afbe56b47ad0442850/lib/Dialect/Torch/Transforms/RefineTypes.cpp#L984-L992

After the decomposition the `AtenEmptyMemoryFormatOp` makes use of `resultType.getElementType()` for the dtype. Whatever we pass from here makes no difference.

Reply:
Right, but the use of `resultType.getElementType()` in `AtenEmptyMemoryFormatOp` is an implementation detail of that op, not a property of it. In general it is not a good idea to depend on implementation details of other functions because this creates a coupling that can make things hard to understand/debug. 

For example, someone in the future could change the `AtenEmptyMemoryFormatOp` to use the `op.dtype()` argument rather than the `resultType` (which would actually be a better way of handling the lowering to decrease the coupling with `RefineTypes`). This would leave the `AtenEmptyMemoryFormatOp` lowering correct, but now the lowering for for `new_empty/ones/zeros` are broken, which is not a good thing. Correct transformations of one part of the code should not break other parts of the code.